File: 3309-count-prefix-and-suffix-pairs-i/3309-count-prefix-and-suffix-pairs-i.cpp
class Solution {
public:
    int countPrefixSuffixPairs(vector<string>& words) {
        int res = 0;

        for (size_t i = 0; i < words.size(); i++) {
            for (size_t j = i + 1; j < words.size(); j++) {
                if (words[j].substr(0, words[i].size()) == words[i] &&
                    words[j].substr(words[j].size() - words[i].size()) == words[i]) {
                    res++;
                }
            }
        }

        return res;
    }

};

File: 3309-count-prefix-and-suffix-pairs-i/3309-count-prefix-and-suffix-pairs-i.py
class Solution:
    def countPrefixSuffixPairs(self, words: List[str]) -> int:
        res = 0
        for i in range(len(words)):
            for j in range(i + 1, len(words)):
                if words[j].startswith(words[i]) and words[j].endswith(words[i]):
                    res += 1
        return res